etna/models/seasonal_ma.py

```diff
@@ -91,14 +92,52 @@ def forecast(self, df: pd.DataFrame, prediction_size: int) -> np.ndarray:
             )
 
         history = df["target"][-expected_length:-prediction_size]
+        if np.any(history.isnull()):
+            raise ValueError("There are NaNs in a forecast context, forecast method required context to filled!")
+
         res = np.append(history, np.zeros(prediction_size))
         for i in range(self.shift, len(res)):
             res[i] = res[i - self.shift : i : self.seasonality].mean()
         y_pred = res[-prediction_size:]
         return y_pred
 
-    def predict(self, df: pd.DataFrame, prediction_size: int) -> pd.DataFrame:
-        raise NotImplementedError("Method predict isn't currently implemented!")
+    def predict(self, df: pd.DataFrame, prediction_size: int) -> np.ndarray:
+        """Compute predictions using true target data as context.
+
+        Parameters
+        ----------
+        df:
+            Used only for getting the horizon of forecast
+        prediction_size:
+            Number of last timestamps to leave after making prediction.
+            Previous timestamps will be used as a context for models that require it.
+
+        Returns
+        -------
+        :
+            Array with predictions.
+
+        Raises
+        ------
+        ValueError:
+            if context isn't big enough
+        ValueError:
```

Comment:
In the future we plan to use this method to fill the nans with this method, how will we handle it if this method fails?